doc/en/backwards-compatibility.rst

```diff
@@ -3,6 +3,68 @@
 Backwards Compatibility Policy
 ==============================
 
+.. versionadded: 6.0
+
+pytest is actively evolving and is a project that has been decades in the making,
+we keep learning about new and better structures to express different details about testing.
+
+While we implement those modifications we try to ensure easy transition and don't want to impose unnecessary churn on our users and community/plugin authors.
+
+As of now, pytest considers multipe types of backward compatibility transitions:
+
+**a) trivial**: APIs which trivially translate to the new mechanism,
+   and do not cause problematic changes.
+
+   We try to support those indefinitely while encouraging users to switch to newer/better mechanisms through documentation.
+
+**b) transitional**: the old and new API don't conflict
+   and we can help users transition by using warnings, while supporting both for a prolonged time.
```

Comment:
This renders a little strangely in GitHub:

![image](https://user-images.githubusercontent.com/1324225/69936491-a31c6b80-14e0-11ea-8338-07f2e4d5bbc8.png)

https://github.com/RonnyPfannschmidt/pytest/blob/deprecation-breakage/doc/en/backwards-compatibility.rst#backwards-compatibility-policy